### PR TITLE
[IOTDB-2222] bugs of Spark Connector

### PR DESCRIPTION
**Pull request by xuanronaldo**

### CODE REVIEW

File: spark-iotdb-connector/src/main/scala/org/apache/iotdb/spark/db/DataFrameTools.scala [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+import org.apache.iotdb.session.Session
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType.{BOOLEAN, DOUBLE, FLOAT, INT32, INT64, TEXT}
+import org.apache.spark.sql.DataFrame
+
+import java.util
+import java.lang
+
+object DataFrameTools {
```
Comment (SilverNarcissus) [inline spark-iotdb-connector/src/main/scala/org/apache/iotdb/spark/db/DataFrameTools.scala:30]: We need some tests of the method.

**Review by SilverNarcissus — COMMENTED** [review]
Brilliant work!  There are failing tests and only one issue, please have a look.

**Review by SilverNarcissus — APPROVED** [review]

File: docs/UserGuide/Ecosystem Integration/Spark IoTDB.md [hunk 2]
```diff
@@ -41,16 +41,15 @@ mvn clean scala:compile compile install
...
 ```
 
-## Read Data from IoTDB
 
-### spark-shell user guide
+#### spark-shell user guide
 
 ```
-spark-shell --jars spark-iotdb-connector-0.13.0-SNAPSHOT.jar,iotdb-jdbc-0.13.0-SNAPSHOT-jar-with-dependencies.jar
+spark-shell --jars spark-iotdb-connector-0.12.4.jar,iotdb-jdbc-0.12.4-jar-with-dependencies.jar
```
Comment (HTHou) [inline docs/UserGuide/Ecosystem Integration/Spark IoTDB.md:52]: 0.12.4?

File: docs/UserGuide/Ecosystem Integration/Spark IoTDB.md [hunk 3]
```diff
@@ -64,7 +63,7 @@ df.show()
 To partition rdd:
 
 ```
-spark-shell --jars spark-iotdb-connector-0.13.0-SNAPSHOT.jar,iotdb-jdbc-0.13.0-SNAPSHOT-jar-with-dependencies.jar
+spark-shell --jars spark-iotdb-connector-0.12.4.jar,iotdb-jdbc-0.12.4-jar-with-dependencies.jar
```
Comment (HTHou) [inline docs/UserGuide/Ecosystem Integration/Spark IoTDB.md:66]: 0.12.4

**Review by HTHou — APPROVED** [review]

**Pull request merged**